Apex Enterprise Pattern/ShoppingCart_Controller.cls

```diff
@@ -0,0 +1,788 @@
+public class ShoppingCart_Controller
```

Comment:
this file is large because every thing is in controller itself
at a glance what I can see is 
- variable naming needs to improve
- it feels like too many redundant method are there
- too much extra work is done for pagination